torch/testing/_internal/distributed/rpc/dist_autograd_test.py

```diff
@@ -1932,3 +1932,80 @@ def backward(ctx, grad):
             self.assertFalse(p_a == p_b)
             # check one of them is using the computed buffer
             self.assertTrue(p_a == p_g or p_b == p_g)
+
+    @dist_init
+    def test_no_grad_copy_sparse(self):
```

Comment:
It's not so nice that the test has to be copy pasted like this. It would be good if at least the two versions of the test are cross-linked from each other.

Reply:
I originally thought about doing this, but the distributed autograd tests are different enough (with the context, a different API and the need to use get_gradients for the grads map) that I wasn't sure how much we would be able to dedup effectively.

Additionally since we removed the optimization of the same grad tensor being used in different variables, the behavior between distributed autograd and local autograd for the same test is slightly different.